py/objmodule.h

```diff
@@ -35,10 +35,20 @@
 #endif
 
 extern const mp_map_t mp_builtin_module_map;
-extern const mp_map_t mp_builtin_extensible_module_map;
 
-mp_obj_t mp_module_get_builtin(qstr module_name, bool extensible);
+#if MICROPY_HAVE_REGISTERED_EXTENSIBLE_MODULES
+extern const mp_map_t mp_builtin_extensible_module_map;
+#endif
 
+#if MICROPY_HAVE_REGISTERED_EXTENSIBLE_MODULES
+mp_obj_t mp_module_get_builtin_impl(qstr module_name, bool extensible);
+#define mp_module_get_builtin(module_name, extensible) mp_module_get_builtin_impl((module_name), (extensible))
+#else
+mp_obj_t mp_module_get_builtin_impl(qstr module_name);
+#define mp_module_get_builtin(module_name, extensible) ( \
+    MP_STATIC_ASSERT(!extensible), mp_module_get_builtin_impl((module_name)) \
+    )
+#endif
```

Comment:
This looks OK.  An alternative (not saying it would be better this way!) is:
```c
#if MICROPY_HAVE_REGISTERED_EXTENSIBLE_MODULES
mp_obj_t mp_module_get_builtin_impl(qstr module_name, bool extensible);
#define mp_module_get_builtin(module_name) mp_module_get_builtin_impl((module_name), false)
#define mp_module_get_builtin_extensible(module_name) mp_module_get_builtin_impl((module_name), true)
#else
mp_obj_t mp_module_get_builtin(qstr module_name);
#endif
```
then change all uses of `mp_module_get_builtin(module, false)` to `mp_module_get_builtin(module)` and change `mp_module_get_builtin(module, true)` to `mp_module_get_builtin_extensible(module)`.

Reply:
Implemented in its own commit, so you can take it or not at your preference.